contrib/golang/filters/http/source/go/pkg/http/capi_impl.go

```diff
@@ -113,6 +113,21 @@ func capiStatusToErr(status C.CAPIStatus) error {
 	return errors.New("unknown status")
 }
 
+func goHeadersToCHeaders(headers map[string][]string) ([]string, func()) {
```

Comment:
Better add a note: This require go1.22 which include https://go-review.googlesource.com/c/go/+/527156

Reply:
I'm not sure it's https://go-review.googlesource.com/c/go/+/527156 since what it being pinned is garanteed to be go pointers in this case no ? Maybe you are reffering to another commit ?

Reply:
maybe was it this https://go-review.googlesource.com/c/go/+/367296 ?

Reply:
nope, https://go-review.googlesource.com/c/go/+/527156 is required, since golang runtime will panic when the pinned pointer is a non-Go pointer, i.e. a const string.
That's why I created https://go-review.googlesource.com/c/go/+/527156.

Reply:
I get it now. `unsafe.StringData` is not garanteed to return a go pointer as it can be a c allocated byte array is the string was constructed as a 'view' on a c allocated memory.